mongo/bulk_write.go

```diff
@@ -151,9 +151,14 @@ func (bw *bulkWrite) runBatch(ctx context.Context, batch bulkWriteBatch) (BulkWr
 	batchErr.WriteErrors = make([]BulkWriteError, 0, len(writeErrors))
 	convWriteErrors := writeErrorsFromDriverWriteErrors(writeErrors)
 	for _, we := range convWriteErrors {
+		model := batch.models[0]
+		if len(batch.models) > we.Index {
```

Comment:
Is this check necessary? Is it possible to have a case where the index is nit valid for `batch.models`?

Reply:
i don't think so, but as I could not find anything about that on the documentation, i prefered to be safer

Reply:
The index should always be the index in the current batch, so I think it's fine to have `batch.models[we.Index]` unconditionally.

Reply:
all right, i made another comment describing my steps to test that